sparse/_coo/core.py

```diff
@@ -2186,6 +2186,9 @@ def astype(self, dtype, copy=True):
         :obj:`COO.elemwise`: Apply an arbitrary element-wise function to one or two
             arguments.
         """
+        # this matches numpy's behavior
+        if self.dtype == dtype and copy == False:
```

Comment:
Comparing dtypes can be tricky. To be safe:

```
if not copy and self.dtype.type == np.dtype(dtype).type:
    return self
```

Reply:
Your suggestion doesn't match the behavior of regular numpy arrays.

Worse, it does the wrong thing for reversed-endian types.

Reply:
Good point. I'll amend my recommendation to only coerce the user input, then:

```
if not copy and self.dtype == np.dtype(dtype):
```

Reply:
That coercion happens automatically anyway, and is probably faster at the C level where it doesn't need a python function call.